datadog/dogstatsd/route.py

```diff
@@ -0,0 +1,32 @@
+"""
+Helper(s), resolve the system's default interface.
+"""
+# stdlib
+import socket
+import struct
+
+
+def get_default_route():
+    """
+    Return the system default interface using the proc filesystem.
+
+    Returns:
+        string: default route
+
+    Raises:
+        `NotImplementedError`: No proc filesystem is found (non-Linux systems)
+        `StopIteration`: No default route found
```

Comment:
The `get_default_route` method can be considered as an iterator on the `/proc/net/route` file: it keeps going  until a "match" is found, i.e.
```python
for iterator:
   if condition:
      return 
```

Considering your comment, I believe this might not be a good choice. My goal was to to stick with a Python builtin exception and this one seemed to be the best candidate .